components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,767 @@
+            return newEditMode;
+        })
+
+        setSlot(props.slot)
+    }, [props.slot])
+
+    useEffect(() => {
+        setErrorMsg(slot.errorMsg);
+    }, [slot])
+
+    const key = `assignee-${props.slot.id}`;
+
+    const renderIdentifier = (assigneeIdentifier: AssigneeIdentifier | undefined) => {
+        if (!assigneeIdentifier) {
+            return undefined;
+        }
+        return `${assigneeIdentifier.identity.authHost}/${assigneeIdentifier.identity.authName}`;
+    };
+
+    const handleAssignment = () => {
+        if (slot.state === 'assigned') {
+            props.inputHandler.reassign(slot, assigneeIdentifier || '');
+        } else {
+            props.inputHandler.assign(slot, assigneeIdentifier || '');
+        }
+    };
+
+    const handleCancel = () => {
+        setEditMode(false);
+        setErrorMsg(undefined);
+        setAssigneeIdentifier(undefined);
+    };
+
+    const handleEdit = () => {
+        setEditMode(true);
+        setErrorMsg(undefined);
+        setAssigneeIdentifier('');
+    };
+
+    const handleDeactivation = () => {
+        props.inputHandler.deactivate(slot);
+    }
+
+    const handleReactivation = () => {
+        props.inputHandler.reactivate(slot);
+    }
+
+
+    const getActions = (slot: Slot) => {
+        const actions: JSX.Element[] = [];
+
+        if (editMode) {
+            actions.push((<button key="ok" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleAssignment()}>OK</button>));
+            actions.push((<button key="cancel" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleCancel()}>Cancel</button>));
+        } else {
+            const actionDeactivate = (<button key="deactivate" className={"ml-2 disabled:opacity-50 border-red-900 bg-red-500 hover:bg-red-700"} disabled={slot.loading} onClick={() => handleDeactivation()}>Deactivate</button>);
+            const actionReAssign = (<button key="reassign" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleEdit()}>Edit</button>);
+            const actionReactivate = (<button key="reactivate" className={"ml-2 disabled:opacity-50"} disabled={slot.loading} onClick={() => handleReactivation()}>Reactivate</button>);
+            switch (slot.state) {
+                case 'unassigned':
+                    actions.push(actionReAssign);
+                    actions.push(actionDeactivate);
+                    break;
+
+                default:
+                case 'assigned':
+                    actions.push(actionReAssign);
+                    actions.push(actionDeactivate);
+                    break;
+
+                case 'deactivated':
+                    actions.push(actionReAssign);
+                    actions.push(actionReactivate);
+                    break;
+            }
+        }
+        return actions;
+    }
+
+    return (
+        <div key={key} className="flex flex-col space-y-2">
```

Comment:
**issue**: We'll need some more space above each label to make each group (label and input) distinct. For reference, see _Gestalt Principles_ [used in design](https://en.wikipedia.org/wiki/Gestalt_psychology#Use_in_design).

```suggestion
        <div key={key} className="flex flex-col space-y-2 pt-2">
```